src/server/string_family.cc

```diff
@@ -480,14 +486,7 @@ void StringFamily::Set(CmdArgList args, ConnectionContext* cntx) {
     }
   }
 
-  DCHECK(cntx->transaction);
-
-  auto cb = [&](Transaction* t, EngineShard* shard) {
-    SetCmd sg(t->GetOpArgs(shard));
-    return sg.Set(sparams, key, value);
-  };
-  OpResult<void> result = cntx->transaction->ScheduleSingleHop(std::move(cb));
-
+  const auto result{SetGeneric(cntx, std::move(sparams), key, value)};
```

Comment:
why moving SetParams and and not using `const ref&` ?

Reply:
Maybe to mach time with Rust :), but in this case we're not really need it at this point any more